### PR TITLE
feat(s3_eye): Support I2C Driver-NG

### PR DESCRIPTION
**Pull request by tore-espressif**
# ESP-BSP Pull Request checklist

## TODO

- [x] Update QMA6100P driver to I2C driver-NG

## Change description
Update ESP32-S3-EYE to I2C Driver-NG

### CODE REVIEW

**Review by espzav — APPROVED** [review]
LGTM, not tested on HW

File: .build-test-rules.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ examples:
       reason: Example depends on BSP, which is supported only for IDF >= 5.2
     - if: (IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 3) and CONFIG_NAME in ["esp32_p4_function_ev_board", "esp-box-3"]
       reason: Example depends on BSP, which is supported only for IDF >= 5.3
-    - if: (IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 4) and CONFIG_NAME in ["m5stack_core_s3", "esp32_s2_kaluga_kit"]
+    - if: (IDF_VERSION_MAJOR == 5 and IDF_VERSION_MINOR < 4) and CONFIG_NAME in ["m5stack_core_s3", "esp32_s2_kaluga_kit", "esp32_s3_eye"]
       reason: Example depends on BSP, which is supported only for IDF >= 5.4
```
Comment (tore-espressif) [inline .build-test-rules.yml:12]: You need to add  IDF version restriction to qma test_app (line 90)

File: components/qma6100p/idf_component.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
-version: "1.0.0"
+version: "2.0.0"
 description: I2C driver for QMA6100P accelerometer
 url: https://github.com/espressif/esp-bsp/tree/master/components/qma6100p
 dependencies:
-  idf : ">=4.0"
+  idf : ">=5.4"
```
Comment (tore-espressif) [inline components/qma6100p/idf_component.yml:5]:
```suggestion
  idf : ">=5.2"
```

Driver-NG was added in idf v5.2

File: components/qma6100p/test_apps/main/idf_component.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ## IDF Component Manager Manifest File
 dependencies:
-  idf: ">=4.0"
+  idf: ">=5.4"
```
Comment (tore-espressif) [inline components/qma6100p/test_apps/main/idf_component.yml:3]:
```suggestion
  idf: ">=5.2"
```
Driver-NG was added in idf v5.2

File: components/qma6100p/qma6100p.c [hunk 4]
```diff
@@ -60,65 +63,54 @@ typedef struct {
...
+    esp_err_t ret = ESP_OK;
+
     qma6100p_dev_t *sensor = (qma6100p_dev_t *) calloc(1, sizeof(qma6100p_dev_t));
-    sensor->bus = port;
-    sensor->dev_addr = dev_addr << 1;
-    sensor->counter = 0;
-    sensor->dt = 0;
-    sensor->timer = (struct timeval *) calloc(1, sizeof(struct timeval));
-    return (qma6100p_handle_t) sensor;
+    struct timeval *timer = (struct timeval *) calloc(1, sizeof(struct timeval));
```
Comment (tore-espressif) [inline components/qma6100p/qma6100p.c:86]: You must assign the `timer` to `sensor->timer`

Comment (tore-espressif) [reply components/qma6100p/qma6100p.c:86]: @pborcin

**Pull request merged**